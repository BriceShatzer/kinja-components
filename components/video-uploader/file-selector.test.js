import React from 'react';
import { shallow, mount } from 'enzyme';

import FileSelector from './file-selector';

const mockFile = new File('abc'.split(''), 'hello.mp4', {
	type: 'video/mp4',
	lastModified: 1529951657565
});

const mockSubmit = jest.fn();

const stubElement = ({
	file = null
} = {}) => (
	<FileSelector
		title="Upload a video"
		file={file}
		onSubmit={mockSubmit}
	/>
);

describe('<VideoUploader />', () => {
	it('should render with file selector', () => {
		const wrapper = shallow(stubElement());
		expect(wrapper).toMatchSnapshot();
	});

	it('should accept a file as props', () => {
		const wrapper = shallow(stubElement({ file: mockFile }));
		expect(wrapper.state('file')).toBe(mockFile);
		expect(wrapper).toMatchSnapshot();
	});

	it('should set file when input changes', () => {
		const wrapper = mount(stubElement());
		wrapper.find('input[type="file"]')
			.simulate('change', { target: { files: [mockFile]}});
		expect(wrapper.state('file')).toBe(mockFile);
	});

	it('should pass the file through "onSubmit" if insert button clicked', () => {
		const wrapper = shallow(stubElement({ file: mockFile }));
		wrapper.find('Button').simulate('click');
		expect(mockSubmit).toHaveBeenCalledWith(mockFile);
	});
});