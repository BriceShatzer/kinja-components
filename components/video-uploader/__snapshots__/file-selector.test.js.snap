// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoUploader /> should accept a file as props 1`] = `
<div>
  <file-selector__Headline>
    Upload a video
  </file-selector__Headline>
  <Filefield
    name="file"
    onChange={[Function]}
  />
  <file-selector__ActionButtonGroup>
    <CancelButton
      label="Cancel"
      onClick={[Function]}
      small={true}
      weight="secondary"
    />
    <Button
      disabled={false}
      label="Insert"
      onClick={[Function]}
      small={true}
      weight="primary"
    />
  </file-selector__ActionButtonGroup>
</div>
`;

exports[`<VideoUploader /> should render with file selector 1`] = `
<div>
  <file-selector__Headline>
    Upload a video
  </file-selector__Headline>
  <Filefield
    name="file"
    onChange={[Function]}
  />
  <file-selector__ActionButtonGroup>
    <CancelButton
      label="Cancel"
      onClick={[Function]}
      small={true}
      weight="secondary"
    />
    <Button
      disabled={true}
      label="Insert"
      onClick={[Function]}
      small={true}
      weight="primary"
    />
  </file-selector__ActionButtonGroup>
</div>
`;
